### PR TITLE
dockerize-job-ad-service-and-depended-services: refactoring of docker configuration

### PR DESCRIPTION
**Pull request by wwebec**
…

### CODE REVIEW

File: docker/jobad/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+         - 2181:2181
+    kafka:
+       image: wurstmeister/kafka:1.0.0
+       environment:
+           KAFKA_ADVERTISED_HOST_NAME: localhost
+           KAFKA_ADVERTISED_PORT: 9092
+           KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+       ports:
+           - 9092:9092
+    keycloak:
```
Comment (pavax) [inline docker/jobad/docker-compose.yml]: why do we need keycloak?

File: docker/jobad/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+       volumes:
+         - ./realm-config:/opt/jboss/keycloak/realm-config
+       environment:
+         - KEYCLOAK_USER=admin
+         - KEYCLOAK_PASSWORD=admin
+       ports:
+         - 9080:9080
+         - 9443:9443
+         - 10990:10990
+    spring-cloud-dataflow:
```
Comment (pavax) [inline docker/jobad/docker-compose.yml]: why do we need spring-cloud-data-flow?

File: docker/jobad/docker-compose.yml (hunk 1 above)
Comment (pavax) [reply docker/jobad/docker-compose.yml]: remove

File: docker/jobad/docker-compose.yml (hunk 2 above)
Comment (pavax) [reply docker/jobad/docker-compose.yml]: not needed here but have a look at the candidate-service

Comment (pavax) [reply docker/jobad/docker-compose.yml]: can we use it like this: 

```
services:
    kafka:
        image: wurstmeister/kafka:1.1.0
        expose:
        - "9092"
        environment:
        - KAFKA_ADVERTISED_PORT=9092
        - KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181
        - KAFKA_ADVERTISED_HOST_NAME=kafka
        depends_on:
        - zookeeper
    zookeeper:
        image: wurstmeister/zookeeper
        expose:
        - "2181"
    dataflow-server:
        image: springcloud/spring-cloud-dataflow-server-local:1.6.1.RELEASE
        container_name: dataflow-server
        ports:
        - "9393:9393"
        - "9000-9010:9000-9010"
        environment:
        - spring.cloud.dataflow.applicationProperties.stream.spring.cloud.stream.kafka.binder.brokers=kafka:9092
        - spring.cloud.dataflow.applicationProperties.stream.spring.cloud.stream.kafka.binder.zkNodes=zookeeper:2181
        depends_on:
        - kafka
        volumes:
        - ~/.m2:/root/.m2
```

File: docker/jobad/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        environment:
+        - POSTGRES_USER=jobadservice
+        - POSTGRES_PASSWORD=
+        ports:
+        - 5434:5432
+    zookeeper:
+       image: wurstmeister/zookeeper:3.4.6
+       ports:
+         - 2181:2181
+    kafka:
```
Comment (pavax) [inline docker/jobad/docker-compose.yml]: also integrate the kafka binding to the application-local.properties BUT still of course disable it (use the mock as it is)

File: docker/jobad/docker-compose.yml (hunk 1 above)
Comment (wwebec) [reply docker/jobad/docker-compose.yml]: removed

File: docker/jobad/docker-compose.yml (hunk 2 above)
Comment (wwebec) [reply docker/jobad/docker-compose.yml]: moved to separate script under ```docker/dataflow/docker-compose.yml```

File: .travis.yml [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
 - docker
+language: java
 jdk:
 - oraclejdk8
 sudo: false
 cache:
   directories:
-  - "$HOME/.gradle"
+  - "$HOME/.gradle/wrapper"
+  - "$HOME/.gradle/caches/modules-2/files-2.1"
```
Comment (birom) [inline .travis.yml]: Hi @wwebec, what is happening here exactly? Where does this directory name come from?

File: gradle/docker-compose.gradle [hunk 5]
```diff
@@ -0,0 +1,9 @@
+apply plugin: 'docker-compose'
```
Comment (birom) [inline gradle/docker-compose.gradle]: We also have a ``docker.gradle`` used root ``build.gradle``. The ``kafka.yml`` and ``registry.yml`` are referenced there, but you deleted these compose files. What is the intention here?

File: docker/dataflow/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+version: '3'
```
Comment (birom) [inline docker/dataflow/docker-compose.yml]: Why do we have sub directories e.g: dataflow, jobad? Would not be more convenient having a flat directory structure?

File: gradle/docker-compose.gradle (hunk 5 above)
Comment (wwebec) [reply gradle/docker-compose.gradle]: In this case ```docker-compose.gradle``` overtakes responsibilty of ```docker.gradle```. 
Removed ```docker.gradle``` and replaced its references with ```docker-compose.gradle```.

File: .travis.yml (hunk 4 above)
Comment (wwebec) [reply .travis.yml]: the directories have changed based on a  ```travis.yml```  of candidate-service.  Restored to the previous version.

File: docker/dataflow/docker-compose.yml (hunk 6 above)
Comment (birom) [reply docker/dataflow/docker-compose.yml]: Ok. Please create a 3rd docker-compose file for kafka and move zookeeper and kafka there (docker/kafka/docker-compose.yml). 
The dataflow server currently is an optional component. We use it only for executing and monitoring the batch applications (x28 import, export). Kafka and zookeeper are needed by the source, sink and the job-ad-service application (if we do not use the avam-mock profile).

Comment (wwebec) [reply docker/dataflow/docker-compose.yml]: done

**Pull request merged**